code/modules/ruins/objects_and_mobs/gym.dm

```diff
@@ -5,18 +5,21 @@
 	icon_state = "punchingbag"
 	anchored = TRUE
 	layer = WALL_OBJ_LAYER
+	var/cooldown = 1 SECONDS
 	var/list/hit_sounds = list('sound/weapons/genhit1.ogg', 'sound/weapons/genhit2.ogg', 'sound/weapons/genhit3.ogg',\
 	'sound/weapons/punch1.ogg', 'sound/weapons/punch2.ogg', 'sound/weapons/punch3.ogg', 'sound/weapons/punch4.ogg')
 
 /obj/structure/punching_bag/attack_hand(mob/user as mob)
+	if(cooldown >= world.time)
+		to_chat(user, "<span class='warning'>The bag is still swinging!</span>")
+		return
+	cooldown = world.time + 1 SECONDS
```

Comment:
instead of doing this, set the users nextmove, using something like `user.changeNext_move(CLICK_CD_MELEE)` that way its in line with all other click cooldowns.